aggregator/src/test/java/org/jenkinsci/plugins/workflow/WorkflowRunTest.java

```diff
@@ -232,4 +230,34 @@ public void failedToStartRun() throws Exception {
         r.assertBuildStatusSuccess(r.waitForCompletion(b));
     }
 
+    @Issue("JENKINS-29571")
+    @Test public void buildRecordAfterRename() throws Exception {
```

Comment:
@jglick The following is a tad rant. It is a personal opinion of mine, feel free to ignore if you want.

I really wish you could stop with your crazy fetish of putting annotations on the same line as the method declaration.

The annotations target the method, that we have put semantic value on some annotations as somehow applying to the return type is not quite correct in my view.

When you have a `public @CheckForNull Object getSomething()` method the return type is not `@CheckForNull Object`...

I do not see you annotating your local variables

```
  public void doMagic() {
    ...
    @CheckForNull Object s = getSomething();
    ...
    if (s != null) {
      @Nonnull Object sth = s;
      ...
    }
    ...
    if (s != null) {
      @Nonnull Object sth = s;
      ...
    }
    ...
  }
```

A method that is annotated `@CheckForNull` is not semantically the same as a method returning `Optional<T>`.

To my thinking, you are annotating the *method* as potentially returning a `null` the type of the return value doesn't magically morph from a `@CheckForNull T` into a `@Nonnull T` and in fact, as soon as you capture that value into a local variable without a `@CheckForNull` or `@Nonnull` annotation what you have effectively done is "convert" the value into a `@Nullable`... only that findbugs does call path tracing for you allows findbugs to upscale that into either a  `@CheckForNull` or `@Nonnull` based on the properties of the *methods* you have called.

Is this another NetBeans has support for differentiating `@TypeQualifier` annotated annotations and formatting them next to the type while retaining the non `@TypeQualifier` annotations on the preceding lines thingy much like the java 8 for tests thingy?

We will ignore for now that JSR 305 is completely stalled, that there has not been a formal release (even a draft) of the annotations from that JSR and that you are not supposed to populate the `javax.*` namespace outside of the JSR process... and we will pretend that `@TypeQualifier` will actually come out of that JSR is it ever resurrects itself (That JSR and Jenkins having chosen to use it's annotations - incorrectly in my view - is a whole different discussion)

I have checked IntelliJ's code formatting options:
<img width="1037" alt="screen shot 2015-07-24 at 10 42 25" src="https://cloud.githubusercontent.com/assets/209336/8871977/4ff7e636-31f1-11e5-825c-0b1c6b5bba9c.png">

No option to put `@TypeQualifier` annotations in between the `(public|protected|private)\s(static\s)?(final\s)?` and the return type. The IDE forced that the type annotations should be before the `(public|protected|private)\s(static\s)?(final\s)?` and my choice is when to line wrap.

Eclipse does not support your code style either:

<img width="990" alt="screen shot 2015-07-24 at 10 45 40" src="https://cloud.githubusercontent.com/assets/209336/8872108/34f7a348-31f2-11e5-95e8-6851df348780.png">

In fact Eclipse is more restrictive. You just get the choice of a new line or not.

After sullying my machine with a temporary Eclipse installation, I said what the heck and installed NetBeans... I could not find any support for your code style:

<img width="755" alt="screen shot 2015-07-24 at 10 52 46" src="https://cloud.githubusercontent.com/assets/209336/8872144/654e81c4-31f2-11e5-8d21-c9391437aa95.png">

So it would seem that the IDEs have the following options:

* Annotations on the same line *but* before the `(public|protected|private)` (All IDEs)
* Annotations on the same line *but* before the `(public|protected|private)` and wrap if long (IntelliJ & NetBeans)
* Annotations on the same line *but* before the `(public|protected|private)` and chop down if long (IntelliJ)
* Annotations, each on their own line, before the line with `(public|protected|private)` (All IDEs)

So none of the IDEs - from what I can see - support your code style. 

That to me is rather annoying. I like when there is a reasonable code style to keep with. 

Historically the Jenkins code style has tended to be "The default code style that ships with IntelliJ"... largely as a result of KK's preferred IDE. Though I appreciate that specific code style is not very friendly to users of other IDEs. 

As the community has grown, each plugin maintainer is free to define their own code style. 

Issues of code style can affect the ability of others to contribute code.

If a plugin maintainer does not provide code style guidelines, it raises the bar for contribution.

To my way of thinking, the best way to encourage contributions is:

* Provide IDE configurations for code formatting for all three major Java IDEs
* (Optional) provide a configuration file for a CLI formatting tool
* Describe how to handle formatting in pull requests (no harm stating the obvious, e.g. do not mix formatting changes with logic changes. For logic changes sanity proof commits with `git add -p` to ensure only the logic change lines are included)
* Describe what spelling conventions to use for new names (e.g. in your case you would say "prefer synch to sync" ;-) )

I recognise that there is a big ask there and it probably will take time to get there... but right now I do not see your current code style as being one that is compatible with any IDEs formatter. That makes Stephen :sob: